### ISSUE TITLE
scvi: discuss big run

### ISSUE THREAD

**Issue by sjfleming** [cellarium-ai/cellarium-ml#333]
# open issues

- [x] run extra tests in pytest
    -  #332 
- [x] HVG selection
    - how many?
        - current thinking: try two runs -- 3000 and 8000
    - should we re-run?
    - [x] need the full list of the relevant glyco genes
- [x] decide on the VAE architecture
    - want to include batch in the encoder at all?  thinking: yes
    - deeply inject covariates?  thinking: yes
    - 128 hidden dims
- [x] decide on model itself
    - [x] create data with "batch" -- should it be [assay, suspension_type, dataset_id, donor_id] ? 
    - do we include categorical_covariates? which ones?
    - current thinking: include categorical_covariates for [assay, suspension_type]
        - conditional on the results of #332 being okay
- [x] decide on batch size (how?)
    - current thinking: go with 512... not too slow, not too huge.  default is 128, lots of ppl do 256... CZI did 1024
- [x] decide on data filtering (how?)
    - 200 UMI counts was used in a previous project where we looked pretty carefully
- [x] decide on data train/test split
    - actually keep some data held out to assess the quality of model checkpoints at various epochs
    - random 1% of the data? would be like 500k cells
    - decision for now: don't hold out anything... we can go outside of cellxgene for validation / test
- [x] decide on KL annealing schedule
    - steps instead of epochs?  20 epochs takes FOREVER to warm up
    - I'm thinking use steps as the increment for KL and anneal it to 1 over a single epoch?  maybe 2?
    - thinking: increment per step over 2 epochs
    - [x] Lys is doing a few runs on a smaller dataset:
        - the above idea
        - no annealing (set KL weight = 1)
        - increment KL weight per epoch over 20 epochs

**Comment by sjfleming** [comment]
Steps:

- run the 3 above KL annealing runs on all the census 2024-07-01 data (no filtering)

**Comment by LysSanzMoreta** [comment]
Extra steps/tasks:

- [ ] fix the predict reconstructions pipeline !!!!!!!

-  [ ] run nn metric against the training data itself (much easier to track than against an external test dataset), i will build pipeline to take trained ckpts , predict latents on all data (makes a small lookup database) and calculate nn against a leave out set. This checks that the training is converging/has converged. I hope to get ~>=70-80% accuracy
- [ ] if we use an external dataset for testing, then it's batch keys need to be modified to match the available ones in the training data, the most worrisome one is the donor_id, the rest they can be matched.
               

 - how important is the donor_id?

- [ ] think about quick testing of the reconstructed values (other than scbi)

**Comment by sjfleming** [comment]
Updated tasks:

- [ ] Stephen: run one_pass_mean_var_std on the full cellxgene_2025_01_30
- [ ] Stephen: compute 8000 highly variable genes
- [ ] Lys: finish testing KL warmup using steps instead of epochs
- [ ] Lys and Stephen: decide on an initial number of epochs to train
- [ ] Lys and Stephen: agree on the config file here #331
    - mainly the same architecture as scvi-tools cellxgene run
- Stephen: run training for 2 models
    - [ ] include batch label in the encoder
    - [ ] do not include batch label in the encoder
